### ISSUE TITLE
Upgrade Web Chat to support DirectLine-Speech in Fairfax

### ISSUE THREAD

**Issue by NickEricson** [microsoft/BotFramework-WebChat#3666]
## Feature Request

Upgrade Web Chat to support DirectLine-Speech in Fairfax.

In Fairfax you need to connect to a different speech service (i.e. the channel endpoint is different). Web Chat should be updated to allow customers to specify the Fx domain so the correct endpoint is used.

cc @trrwilson for details.

[feature-request]

**Comment by corinagum** [comment]
@NickEricson is this a customer request, or planned internally? (for tracking)

**Comment by NickEricson** [comment]
This is planned work. Sorry if I got the Labels wrong.

**Comment by corinagum** [comment]
No problem - they're automatic. Just want to make sure they're accurate. :) (Updating now)

**Comment by compulim** [comment]
This is related to #3377.

I don't recall, but we might need to bump Speech SDK to get the sovereign cloud support, which is #3575.

**Comment by NickEricson** [comment]
@compulim I think this is related / duplicate of #3377. I believe the current sdk supports this via a named parameter and the next sdk will support it officially. @trrwilson can explain how to do this and timelines.

**Comment by trrwilson** [comment]
Thanks, @NickEricson !

Yes, I've recently published a PR to add friendly `fromHost` and `fromEndpoint` factories to `BotFrameworkConfig` in the JavaScript Speech SDK. You can find that here: https://github.com/microsoft/cognitive-services-speech-sdk-js/pull/303

In the interim, it's fully possible to get all of this functionality (albeit it less elegantly) by setting values in the configuration property bag. E.g. if you'd like to use "my.host.convai.azure.us", you can:
```javascript
let config: BotFrameworkConfig = sdk.BotFrameworkConfig.fromSubscription("mySubscriptionKey", "unusedRegion");
config.setProperty(sdk.PropertyId.SpeechServiceConnection_Host, "wss://my.host.convai.azure.us");
let connector: DialogServiceConnector = new DialogServiceConnector(config);
```

**Comment by corinagum** [comment]
Related to #3575

**Comment by trrwilson** [comment]
The aforementioned SDK PR with `fromHost` methods added to the JavaScript `BotFrameworkConfig` object is now in and available for private builds (clone and npm install; npm run build if needed). It'll for sure be included in the 1.15 release, which we're targeting for next week.

Just to reiterate (I'll sound like a broken record, but it's worth it if it helps even a little!): this doesn't make anything that wasn't possible before possible now--it just makes it much more intuitive and convenient to use custom hosts and endpoints vs. needing to set the configuration property manually.

**Comment by NickEricson** [comment]
@trrwilson Thank you for all the details on both how to connect to Fairfax using the current SDK and the updates for the 1.15 SDK.

**Comment by compulim** [comment]
I am bumping to 1.15.0 and it is working. Once the PR is merged, I will provide more information here (or in the PR description).

**Comment by compulim** [comment]
PR is under review now.

To enable sovereign clouds for Direct Line Speech. The `fetchCredentials` option passed to `DirectLineSpeechSDK.createAdapters()` call should return `directLineSpeechHostname`, instead of `region`. For example:

```js
return { authorizationToken: '...', directLineSpeechHostname: 'virginia.convai.speech.azure.us' };
```

This change is outlined in the updated `DIRECT_LINE_SPEECH.md`.
